stan/math/opencl/kernel_generator/multi_result_kernel.hpp

```diff
@@ -576,6 +591,22 @@ class results_cl {
     }
     return std::make_tuple();
   }
+
+  /**
+   * Optimized move assignment of a `matrix_cl` into another `matrix_cl`.
+   * @param result result - check
+   * @param pass bool scalar
+   * @return an empty tuple
+   */
+  template <typename T_result, typename T_matrix,
+            require_all_matrix_cl_t<T_result, T_matrix>* = nullptr,
+            require_all_st_same<T_result, T_matrix>* = nullptr,
+            require_t<std::is_rvalue_reference<T_matrix&&>>* = nullptr>
+  static std::tuple<> make_assignment_pair(T_result&& result,
+                                           T_matrix&& matrix) {
+    result = std::move(matrix);
+    return std::make_tuple();
+  }
```

Comment:
Idt I'm following the logic here, what does this return back an empty tuple? Things like this would be good to add to the docs otherwise it's hard to see what's going on without more context

Reply:
This assignment is implemented by calling `matrix_cl` move constructor, so there is nothing to do in the kernel.